lib/azure/armrest/model/base_model.rb

```diff
@@ -33,10 +33,21 @@ class BaseModel < Delegator
       #
       def initialize(json)
         @json = json
+        @resource_group = nil
         @ostruct = JSON.parse(json, object_class: OpenStruct)
         __setobj__(@ostruct)
       end
 
+      # Return the resource group for the current object.
+      def resource_group
+        @resource_group ||= id[/resourceGroups\/(.+?)\//i, 1] rescue nil
+      end
+
+      # Return a hash of tags associated with the resource.
+      def tags
```

Comment:
If tags is nil it will return an empty hash. I have a spec for that.

I like the idea of a hash_property method.